adding support for shorty k1 keyboard

axshae opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

waffle87 commented on .gitmodules:
Changes to this file should be reverted.

waffle87 commented on keyboards/shorty/k1/keyboard.json:
```diff
@@ -0,0 +1,40 @@
+        "bootmagic": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
NKRO probably doesn't serve much purpose on this keyboard.
```suggestion
        "mousekey": true
```

waffle87 commented on keyboards/shorty/k1/keyboard.json:
Please run `qmk format-json -i keyboards/shorty/k1/keyboard.json` on this file.

waffle87 commented on keyboards/shorty/k1/keyboard.json:
```diff
@@ -0,0 +1,40 @@
+        "vid": "0x5453"
+    },
+    "layouts": {
+        "LAYOUT": {
```
This layout does not very accurately represent the [physical layout](https://hawtkeys.com/product/shorty-k1/) of this board:

![image](https://github.com/user-attachments/assets/c6e629df-f413-40f5-bcf1-da74c43e3d5e)

It's unclear which is the encoder.

axshae replied:
The 4th key at (3,0) represents the encoder. How do you suggest putting that in the layout?

waffle87 commented on keyboards/shorty/k1/readme.md:
```diff
@@ -0,0 +1,27 @@
+# Shorty K1
+
+![Shorty K1 by hawtkeys.com](https://hawtkeys.com/wp-content/uploads/2025/06/12-crop2.png)
```
This image should be rehosted on Imgur.

axshae replied:
Imgur is not working in my country. Imgbb would be fine?

waffle87 commented on keyboards/shorty/zero/keyboard.json:
```diff
@@ -0,0 +1,48 @@
+        "bootmagic": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
```suggestion
        "mousekey": true
```

waffle87 reviewed (changes requested):
Per the PR checklist, PRs should contain the fewest amount of changes (eg. not multiple keyboards). In the future, please keep this in mind.

zvecr commented:
You already have a keyboard at `keyboards/shorty/`. Adding in additional keyboards to this location will require the existing keyboard files to be moved to a different sub folder.

axshae commented:
> You already have a keyboard at `keyboards/shorty/`. Adding in additional keyboards to this location will require the existing keyboard files to be moved to a different sub folder.

I shouldn’t have made so many changes. That was a mistake on my part. I have moved the existing keyboard definition into a folder "shorty/zero" and added a new keyboard definition inside the "shorty/k1" folder.

zvecr commented:
> I have moved the existing keyboard definition into a folder "shorty/zero"

Nope, as can be seen by lint throwing errors and the "files changed" tab showing it as newly added files.

Your base is too out of date, and does not reflect the current state of `qmk/qmk_firmware` and would need to be updated.

axshae commented:
> > I have moved the existing keyboard definition into a folder "shorty/zero"
> 
> Nope, as can be seen by lint throwing errors in CI and the "files changed" tab showing it as newly added files.
> 
> Your base is too out of date, and does not reflect the current state of `qmk/qmk_firmware` and would need to be updated.
> 
> Moving keyboards will also require this PR to go via `develop`.

Thanks for pointing that out. You're right, I missed syncing with the latest upstream before making the changes. I’ll rebase my branch on the current develop and open a new pull request following the proper guidelines.
